### ISSUE TITLE
Null pointer exception on passing empty search attribute in workflow options

### ISSUE THREAD

**Issue by lazy-ninja** [temporalio/sdk-java#2276]
## Expected Behavior
It shouldn't throw a null pointer exception.

```java 
//return null for empty search attributes. 
SearchAttributesUtil.encodeTyped(options.getTypedSearchAttributes()); 

//on setting null search attributes in the StartWorkflowExecutionRequest, it will throw NPE
request.setSearchAttributes(startParameters.getSearchAttributes());

//StartWorkflowExecutionRequest#setSearchAttributes
if (searchAttributesBuilder_ == null) {
  if (value == null) {
    throw new NullPointerException();
  }

```

## Actual Behavior
It throws null pointer exception

## Steps to Reproduce the Problem
  1. Create workflow options as 
  ```java 
WorkflowOptions workflowOptions = WorkflowOptions.newBuilder()
        .setTaskQueue("queue")
        .setWorkflowId("workflow-id-1")
        .setTypedSearchAttributes(SearchAttributes.EMPTY)
        .build();
  ```
 2. create workflow stub for any workflow
 ```java 
stub = workflowClient.newWorkflowStub(SampleWorkflow.class, workflowOptions);
```
3.  Submit for execution 
```java 
WorkflowClient.start(stub::execute, {});
```
## Specifications

  - Version: 1.22.3
  - Platform: MacOS

Let me know if I can raise the PR for it.

**Comment by Quinn-With-Two-Ns** [comment]
Thanks for the bug report! will file a fix for this.

**Comment by lazy-ninja** [comment]
@Quinn-With-Two-Ns thanks for fixing it. can you fix it in the `1.22.x` as well?
